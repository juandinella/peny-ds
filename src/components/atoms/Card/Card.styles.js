import { css } from 'styled-components'

import layout from '../../../styles/theme/layout'

export default css`
  background-color: var(--color-neutral-00);
  width: 100%;
  height: 100%;
  position: relative;

  ${(props) =>
    props.shadow &&
    css`
      box-shadow: var(--shadow-sm);
    `}

  ${(props) =>
    props.borderRadius &&
    css`
      border-radius: var(--radius-${(props) => props.borderRadius});
    `}


  ${(props) => {
    const sizeVerticalMobile = props.mSize && props.mSize.split(' ')[0]

    const sizeHorizontalMobile =
      (props.mSize && props.mSize.split(' ')[1]) ||
      (props.mSize && props.mSize.split(' ')[0])

    const sizeVertical = props.size && props.size.split(' ')[0]

    const sizeHorizontal =
      (props.size && props.size.split(' ')[1]) ||
      (props.size && props.size.split(' ')[0])

    return (
      props.size &&
      css`
        padding: var(
            --spacing-${sizeVerticalMobile ? sizeVerticalMobile : sizeVertical}
          )
          var(
            --spacing-${sizeHorizontalMobile ? sizeHorizontalMobile : sizeHorizontal}
          );

        @media screen and (min-width: ${() => layout.breakpoints.sm}) {
          padding: var(--spacing-${sizeVertical})
            var(--spacing-${sizeHorizontal});
        }
      `
    )
  }}

  ${(props) =>
    props.center &&
    css`
      display: flex;
      justify-content: center;
      align-items: center;
      flex-direction: column;
    `}
`
