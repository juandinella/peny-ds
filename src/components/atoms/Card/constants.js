export const options = {
  /* Size of cards */
  sizes: ['xxs', 'xs', 'sm', 'md', 'lg', 'xl', 'xxl', 'xxxl'],
  /* Border Radius */
  borderRadiuses: ['xs', 'sm', 'lg'],
}
