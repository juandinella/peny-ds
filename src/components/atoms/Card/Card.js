import React from 'react'
import PropTypes from 'prop-types'
import styles from './Card.styles'
import { options } from './constants'

/**
 * Card
 */

const Card = ({
  className,
  id,
  mSize,
  size,
  maxWidth,
  children,
  onClick,
  center,
  shadow,
  borderRadius,
}) => {
  return (
    <div
      css={styles}
      className={className}
      style={{ maxWidth: maxWidth }}
      id={id}
      data-testid={id}
      size={size}
      mSize={mSize}
      center={center}
      onClick={onClick || null}
      shadow={shadow}
      borderRadius={borderRadius}
    >
      {children}
    </div>
  )
}
Card.displayName = 'Card'

Card.propTypes = {
  /** Card's id */
  id: PropTypes.string,
  /** To add style */
  className: PropTypes.string,
  /** Content to be included in the card */
  children: PropTypes.node.isRequired,
  /** Card's size */
  size: PropTypes.oneOf(options.sizes),
  /** Card's size form mobile */
  mSize: PropTypes.oneOf(options.sizes),
  /** Set maxWidth of the container */
  maxWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Click event */
  onClick: PropTypes.func,
  /** Center Children */
  center: PropTypes.bool,
  /** Card's shadow */
  shadow: PropTypes.bool,
  /** Border Rounded */
  borderRadius: PropTypes.oneOf(options.borderRadiuses),
}

Card.defaultProps = {
  shadow: true,
  roundedBorders: true,
}

export default Card
