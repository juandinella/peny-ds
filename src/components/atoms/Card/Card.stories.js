import Card from './Card'
import { options } from './constants'

import {
  getTemplate,
  getListTemplate,
  getOptionsArgTypes,
} from '../../../helpers/storybook'

const Template = getTemplate(Card)
const ListTemplate = getListTemplate(Card)

export default {
  title: 'Atoms/Card',
  component: Card,
  args: {
    children: 'Card',
    size: 'md',
  },
  argTypes: {
    borderRadius: getOptionsArgTypes(options.borderRadiuses),
    size: getOptionsArgTypes(options.sizes),
    mSize: getOptionsArgTypes(options.sizes),
  },
}

export const Default = Template.bind({})

export const Sizes = ListTemplate.bind({})
Sizes.args = { items: options.sizes.map((size) => ({ size })) }

export const Centered = Template.bind({})
Centered.args = { center: true }

export const BorderRadiuses = ListTemplate.bind({})
BorderRadiuses.args = {
  items: options.borderRadiuses.map((borderRadius) => ({ borderRadius })),
}
