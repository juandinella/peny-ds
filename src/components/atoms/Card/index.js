import Card from './Card'

export default Card
export { options } from './constants'
