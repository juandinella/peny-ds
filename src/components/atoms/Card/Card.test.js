import React from 'react'

import { render, cleanup } from '@testing-library/react'

import Card from './Card'

import initStoryshots from '@storybook/addon-storyshots'

initStoryshots({
  storyKindRegex: /^Atoms\/Card$/,
})

const componentProps = {
  id: 'unique_id',
  children: <span data-testid="banner_span">Im a span</span>,
  className: 'custom_class',
}

afterEach(cleanup)

describe('<Card />', () => {
  test('should render component correctly', () => {
    const { queryByTestId } = render(<Card {...componentProps} />)

    const card = queryByTestId(componentProps.id)

    expect(card).toBeTruthy()
  })

  test('should render children correctly', () => {
    const { queryByTestId } = render(<Card {...componentProps} />)

    expect(queryByTestId(componentProps.id).children).toHaveLength(1)

    expect(queryByTestId('banner_span')).toBeTruthy()

    expect(queryByTestId('banner_span').textContent).toEqual('Im a span')
  })

  test('should render class correctly', () => {
    const { queryByTestId, rerender } = render(<Card {...componentProps} />)

    const card = queryByTestId(componentProps.id)

    expect(card.className).toContain('custom_class')

    rerender(<Card {...componentProps} className={null} />)

    expect(card.className).not.toContain('custom_class')
  })
})
