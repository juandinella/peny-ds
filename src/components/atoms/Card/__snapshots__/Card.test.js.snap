// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Card Border Radiuses 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-19ernvp-0 fuLKYX"
    onClick={null}
    size="md"
    style={
      Object {
        "maxWidth": undefined,
      }
    }
  >
    Card
  </div>
  <div
    className="sc-19ernvp-0 cLrbUK"
    onClick={null}
    size="md"
    style={
      Object {
        "maxWidth": undefined,
      }
    }
  >
    Card
  </div>
  <div
    className="sc-19ernvp-0 ilXHZT"
    onClick={null}
    size="md"
    style={
      Object {
        "maxWidth": undefined,
      }
    }
  >
    Card
  </div>
</div>
`;

exports[`Storyshots Atoms/Card Centered 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-19ernvp-0 eIOJyJ"
    onClick={null}
    size="md"
    style={
      Object {
        "maxWidth": undefined,
      }
    }
  >
    Card
  </div>
</div>
`;

exports[`Storyshots Atoms/Card Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-19ernvp-0 fXsjtV"
    onClick={null}
    size="md"
    style={
      Object {
        "maxWidth": undefined,
      }
    }
  >
    Card
  </div>
</div>
`;

exports[`Storyshots Atoms/Card Sizes 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-19ernvp-0 oHKfx"
    onClick={null}
    size="xxs"
    style={
      Object {
        "maxWidth": undefined,
      }
    }
  >
    Card
  </div>
  <div
    className="sc-19ernvp-0 ivkNVF"
    onClick={null}
    size="xs"
    style={
      Object {
        "maxWidth": undefined,
      }
    }
  >
    Card
  </div>
  <div
    className="sc-19ernvp-0 jVzuLx"
    onClick={null}
    size="sm"
    style={
      Object {
        "maxWidth": undefined,
      }
    }
  >
    Card
  </div>
  <div
    className="sc-19ernvp-0 fXsjtV"
    onClick={null}
    size="md"
    style={
      Object {
        "maxWidth": undefined,
      }
    }
  >
    Card
  </div>
  <div
    className="sc-19ernvp-0 kpCQUJ"
    onClick={null}
    size="lg"
    style={
      Object {
        "maxWidth": undefined,
      }
    }
  >
    Card
  </div>
  <div
    className="sc-19ernvp-0 fkkIhp"
    onClick={null}
    size="xl"
    style={
      Object {
        "maxWidth": undefined,
      }
    }
  >
    Card
  </div>
  <div
    className="sc-19ernvp-0 iYcBjx"
    onClick={null}
    size="xxl"
    style={
      Object {
        "maxWidth": undefined,
      }
    }
  >
    Card
  </div>
  <div
    className="sc-19ernvp-0 kEQOXN"
    onClick={null}
    size="xxxl"
    style={
      Object {
        "maxWidth": undefined,
      }
    }
  >
    Card
  </div>
</div>
`;
